src/pages/form/advanced-form/components/TableForm.tsx

```diff
@@ -0,0 +1,297 @@
+import { Button, Divider, Input, Popconfirm, Table, message } from 'antd';
+import { PlusOutlined } from '@ant-design/icons';
+import React, { Fragment, PureComponent } from 'react';
+
+import isEqual from 'lodash.isequal';
+import styles from '../style.less';
+
+interface TableFormDateType {
+  key: string;
+  workId?: string;
+  name?: string;
+  department?: string;
+  isNew?: boolean;
+  editable?: boolean;
+}
+interface TableFormProps {
+  loading?: boolean;
+  value?: TableFormDateType[];
+  onChange?: (value: TableFormDateType[]) => void;
+}
+
+interface TableFormState {
+  loading?: boolean;
+  value?: TableFormDateType[];
+  data?: TableFormDateType[];
+}
+class TableForm extends PureComponent<TableFormProps, TableFormState> {
+  static getDerivedStateFromProps(nextProps: TableFormProps, preState: TableFormState) {
+    if (isEqual(nextProps.value, preState.value)) {
+      return null;
+    }
+    return {
+      data: nextProps.value,
+      value: nextProps.value,
+    };
+  }
+
+  clickedCancel: boolean = false;
+
+  index = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of public instance field not allowed after declaration of public static method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=4583897)